### DISCUSSION TITLE
accessing breadcrumbs metadata from within a dataviewjs codeblock ?

### DISCUSSION THREAD

**Discussion by selimrbd** [SkepticMystic/breadcrumbs#428] [Q&A]
Hey @SkepticMystic 

First of all, thanks a lot for this plugin, its completely transforming the way I can manipulate my files in my vault.

I'd really like to combine breadcrumbs and dataviewjs in a same codeblock, to do things like **creating a table containing all childrens of a node**. Say for example I have the following hierarchy:

- a1
   - a1.1
      - a1.1.1
   - a1.2

I'd like to create in a1 a table that looks like this (and potentially add other columns later):

note    | depth | 
a1.1.    |  1         |
a1.2    |  1         |
a1.1.1  |  2         |

The only way I find this possible for now is adding the "down" metadata to the YAML file using 'breadcrumbs: write to current file' then accessing recursively the file names in a dataviewjs codeblock. Is there a way to do this by accessing directly the breadcrumbs metadata from a dataviewjs file ?

**Comment by SkepticMystic** [comment]
Hey Selim :)

Have a look at the eocs site here: https://breadcrumbs-wiki.onrender.com/docs/Home
Then look into the Breadcrumbs API. It allows you to access BC data anywhere you can run js, so a dataviewjs block will work.

**Comment by selimrbd** [comment]
Here is the solution I came up with, if anyone is looking to do the same thing:

```dataviewjs
// LIST ALL CHILDREN OF THE CURRENT NODE AS WELL AS THEIR DEPTH

function check_if_parent(father, link) {
	if (!link.hasOwnProperty('up')) return false;
	var is_parent = new Boolean(false);
	if (Array.isArray(link.up)) {
			is_parent = link.up.map(x => x.path).includes(father.file.path)
		}
		else {
			is_parent = (link.up.path == father.file.path)
		}
	return is_parent
}

function get_childrens(note, d=1){
	if (!note) return new Set()
	var childrens = new Set()
	for (let inlink of note.file.inlinks.array()) {
		let link = dv.page(inlink.path)
		let is_parent = check_if_parent(note, link)
		if (!is_parent) continue;
		link._depth = d
		childrens.add(link)
		var sub_chds =  get_childrens(link, d+1)
		var childrens = new Set([...childrens, ...sub_chds]);
	}
	return childrens
}

function compare(a, b) {
	// order by depth first, then file name
	return (a._depth - b._depth) || a.file.name.localeCompare(b.file.name)
}

var note = dv.page(dv.current().file.path);
var childrens = get_childrens(note)
var data = dv.array(Array.from(childrens));

dv.table(["note", "depth"], data.values.sort(compare).map(n => [n.file.link, n._depth]));
```

**Reply by MalteRoentgen** [reply]
Hello,
I'm very interested in this! Unfortunately, for me this snippet does not work :-(. I'm admittedly new to Obsidian, is there some very basic thing that I have to do to get the snippet to run? Like, putting a "dataviewjs" in front of the
"```"
? I tried that already, but in this case I get
Dataview: No results to show for table query

Remark: The note in which I have copied the snippet has a child.

Best
Malte

**Reply by selimrbd** [reply]
Hi @MzunguKichaa , yes indeed you need to put "dataviewjs" in the codeblock, just after the first "```".

This code will only work for hierarchical relationships that use the "up" inline metadata. Is that the one you use ?

**Reply by MalteRoentgen** [reply]
Hi @selimrbd, thanks a lot for that clarification! I had my hierarchical relationships defined in the YAML-header, with inline metadata it works now. Thanks a lot!

**Reply by selimrbd** [reply]
Great, happy it was helpful ! 👍

**Reply by icytree** [reply]
Hi guys I am also looking for a way to use Breadcrumb's API but in an function used in a [formula](https://rafaelgb.github.io/obsidian-db-folder/features/Formulas/) in the [Obidian Database Plugin](https://github.com/RafaelGB/obsidian-db-folder).  Any help would be appreciated as I have zero knowledge of how to invoke APIs nor of how to use JavaScript. Here's the [initial discussion](https://github.com/blacksmithgu/obsidian-dataview/discussions/1452) for more details.
